### PR TITLE
Add yamllint config.

### PR DESCRIPTION
**Pull request by kirkrodrigues**
# Description

This config is similar to the one used in [clp](https://github.com/y-scope/clp). A few modifications were made to respect the latest rules.

# Validation performed

Validated CLP's `lint:yml-check` task passed with this config.

### CODE REVIEW

File: lint-configs/.yamllint.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - "*.yml"
+
+rules:
+  anchors:
+    # Disabled so that we can reference aliases that exist later in the file
+    forbid-undeclared-aliases: false
+
+  braces:
+    min-spaces-inside-empty: 0
+    max-spaces-inside-empty: 0
```
Comment (LinZhihao-723) [inline lint-configs/.yamllint.yml:21]: Do we want to ensure alpha-beta ordering?
```suggestion
    max-spaces-inside-empty: 0
    min-spaces-inside-empty: 0
```

File: lint-configs/.yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  brackets:
+    min-spaces-inside-empty: 0
+    max-spaces-inside-empty: 0
```
Comment (LinZhihao-723) [inline lint-configs/.yamllint.yml:25]: Same

File: lint-configs/.yamllint.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  float-values:
+    require-numeral-before-decimal: true
+
+  indentation:
+    spaces: 2
+
+  line-length:
+    max: 100
+    allow-non-breakable-words: false
```
Comment (LinZhihao-723) [inline lint-configs/.yamllint.yml:49]: Same

File: lint-configs/.yamllint.yml (hunk 1 above)
Comment (kirkrodrigues) [reply lint-configs/.yamllint.yml]: Good idea.

**Review by LinZhihao-723 — APPROVED** [review]
commit msg lgtm

**Pull request merged**